scheduler.c

```diff
@@ -59,13 +59,23 @@ VALUE rb_scheduler_io_wait_writable(VALUE scheduler, VALUE io)
     return rb_scheduler_io_wait(scheduler, io, RB_UINT2NUM(RUBY_IO_WRITABLE), Qnil);
 }
 
-VALUE rb_scheduler_io_read(VALUE scheduler, VALUE io, VALUE buffer, VALUE offset, VALUE length)
+int rb_scheduler_supports_io_read(VALUE scheduler)
 {
-    return rb_funcall(scheduler, id_io_read, 4, io, buffer, offset, length);
+    return rb_respond_to(scheduler, id_io_read);
 }
 
-VALUE rb_scheduler_io_write(VALUE scheduler, VALUE io, VALUE buffer, VALUE offset, VALUE length)
+VALUE rb_scheduler_io_read(VALUE scheduler, VALUE io, VALUE buffer, size_t offset, size_t length)
+{
+    return rb_funcall(scheduler, id_io_read, 4, io, buffer, SIZET2NUM(offset), SIZET2NUM(length));
```

Comment:
@shyouhei should we add `RB_SIZET2NUM` etc?

Reply:
Or maybe we can call it `RB_SIZE2NUM` because `SIZET` just looks weird :)

Reply:
Do we need a `OFFT2NUM`, since `offset` could be a negative number?

Reply:
Offset is the offset from the start of the buffer, so it should only ever be positive.